src/spesh/inline.c

```diff
@@ -270,14 +270,18 @@ MVMSpeshGraph * MVM_spesh_inline_try_get_graph(MVMThreadContext *tc, MVMSpeshGra
 
     /* Build graph from the already-specialized bytecode and check if we can
      * inline the graph. */
-    ig = MVM_spesh_graph_create_from_cand(tc, target_sf, cand, 0, &deopt_usage_ins);
+    MVMROOT(tc, cand, {
+        ig = MVM_spesh_graph_create_from_cand(tc, target_sf, cand, 0, &deopt_usage_ins);
+    });
     if (is_graph_inlineable(tc, inliner, target_sf, invoke_ins, ig, no_inline_reason, no_inline_info)) {
         /* We can inline it. Do facts discovery, which also sets usage counts.
          * We also need to bump counts for any inline's code_ref_reg to make
          * sure it stays available for deopt. */
         MVMuint32 i;
-        MVM_spesh_facts_discover(tc, ig, NULL, 1);
-        add_deopt_usages(tc, ig, cand->deopt_usage_info, deopt_usage_ins);
+        MVMROOT(tc, cand, {
+            MVM_spesh_facts_discover(tc, ig, NULL, 1);
+        });
+        add_deopt_usages(tc, ig, cand->body.deopt_usage_info, deopt_usage_ins);
```

Comment:
Pretty sure this doesn't allocate either.

Reply:
@niner thinks it does https://github.com/MoarVM/MoarVM/blob/master/tools/check-roots.py#L230

Reply:
I generated that list with `cflow --number --brief --reverse  src/*.c src/*/*.c src/*/*/*.c  > call_graph` and then extracted the sub graph resulting in a run_gc call from that huge file.

The path it found is MVM_spesh_facts_discover -> add_bb_facts -> wvalfrom_facts -> MVM_sc_get_object -> MVM_serialization_demand_object -> stub_object -> read_object_table_entry -> MVM_sc_get_stable -> MVM_serialization_demand_stable -> stub_stable -> MVM_gc_allocate_stable

But, looking at wvalfrom_facts, the call to MVM_sc_get_object is protected by a MVM_sc_is_object_immediately_available condition. So at least this path is incorrect.

I guess we would need some custom annotations to get a better list. But then, adding unnecessary MVMROOTs hurts less than missing some.

Reply:
Or a list of false-positive-causing functions in check-roots.py? Anyway, removing this one since we've confirmed it's not needed.